generator/ServiceClientGeneratorLib/Generators/Endpoints/EndpointResolver.partial.cs

```diff
@@ -25,7 +25,7 @@ private string GetValueSource(Parameter parameter)
                 case "AWS::S3::DisableMultiRegionAccessPoints": return "config.DisableMultiregionAccessPoints";
                 case "AWS::S3::UseGlobalEndpoint": return "config.USEast1RegionalEndpointValue == S3UsEast1RegionalEndpointValue.Legacy";
                 case "AWS::STS::UseGlobalEndpoint": return "false";
-                case "AWS::Auth::AccountId": return "requestContext.ImmutableCredentials.AccountId";
+                case "AWS::Auth::AccountId": return "(requestContext.Identity as AWSCredentials)?.GetCredentials()?.AccountId";
```

Comment:
Agree we don't need to fix it now. Most of the credential providers will also return cached values if available (so it'll only be expensive occasionally instead of every request).